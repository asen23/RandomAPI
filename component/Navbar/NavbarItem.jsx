import Link from "next/link"

export default function NavbarItem(props) {
    return (
        <Link href={props.href}>
            <a className="px-4 flex justify-center items-center hover:bg-[antiquewhite]">{props.children}</a>
        </Link>
    )
}
