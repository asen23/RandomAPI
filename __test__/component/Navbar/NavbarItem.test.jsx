import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import NavbarItem from "component/Navbar/NavbarItem"

describe("Navbar", () => {
    it("renders unchanged", async () => {
        const { asFragment } = render(<NavbarItem href="/test">A Component</NavbarItem>)
        expect(asFragment()).toMatchSnapshot()
    })

    it("renders its children", () => {
        render(<NavbarItem href="/test">A Component</NavbarItem>)

        const child = screen.getByText("A Component")
        expect(child).toBeInTheDocument()
    })
})
