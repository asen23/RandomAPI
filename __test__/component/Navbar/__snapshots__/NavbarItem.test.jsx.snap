// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders unchanged 1`] = `
<DocumentFragment>
  <a
    class="px-4 flex justify-center items-center hover:bg-primary"
    href="/test"
  >
    A Component
  </a>
</DocumentFragment>
`;
